addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/rest/RestClientImpl.java

```diff
@@ -0,0 +1,144 @@
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the {@link RestClient} interface that is using Jetty HttpClient to perform the requests
+ * and is loading a {@link Certificate} from the OSGi bundle
+ *
+ * @author Svilen Valkanov
+ *
+ */
+public class RestClientImpl implements RestClient {
+
+    private final Logger logger = LoggerFactory.getLogger(RestClientImpl.class);
+
+    private HttpClient httpClient;
+    private BundleContext bundleContext;
+    private String baseURL;
+    private int timeout;
+
+    public void activate(BundleContext bundleContext) throws Exception {
+        this.bundleContext = bundleContext;
+
+        httpClient = new HttpClient(new SslContextFactory());
+
+        loadCertificate(CERTIFICATE_DIR, CERTIFICATE_FILE_NAME);
+        setBaseURL(MIHOME_REST_API_URL);
+        setConnectionTimeout(DEFAULT_REQUEST_TIMEOUT);
+        startHttpClient();
+    }
+
+    public void deactivate(BundleContext context) throws Exception {
+        stopHttpClient();
+    }
+
+    @Override
+    public synchronized ContentResponse sendRequest(String requestPath, HttpMethod httpMethod, Properties httpHeaders,
+            InputStream content, String contentType) throws IOException {
+
+        String url = getBaseURL() + "/" + requestPath;
+        Request request = httpClient.newRequest(url).method(httpMethod).timeout(getConnectionTimeout(),
+                TimeUnit.MILLISECONDS);
+
+        if (httpHeaders != null) {
+            for (String httpHeaderKey : httpHeaders.stringPropertyNames()) {
+                request.header(httpHeaderKey, httpHeaders.getProperty(httpHeaderKey));
+            }
+        }
+
+        // add content if a valid method is given ...
+        if (content != null && (httpMethod.equals(HttpMethod.POST) || httpMethod.equals(HttpMethod.PUT))) {
+            request.content(new InputStreamContentProvider(content), contentType);
+        }
+
+        logger.debug("About to execute {}", request.getURI());
+        ContentResponse response = null;
+        try {
+            response = request.send();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (TimeoutException | ExecutionException e) {
+            throw new IOException(e);
+        }
+        return response;
+    }
+
+    private void loadCertificate(String relativePath, String file) {
+        SSLContext sslContext = SSLContextBuilder.create(bundleContext).withTrustedCertificate(relativePath, file)
+                .build();
+        if (sslContext != null) {
+            httpClient.getSslContextFactory().setSslContext(sslContext);
+        } else {
+            logger.error("Not able to load Mi|Home certificate. The connection with the server may not be secure.");
+        }
+
+    }
+
+    @Override
+    public void setSslContext(SSLContext sslContext) throws Exception {
```

Comment:
Is this method called anywhere? Does it need to be in the interface? Will it handle `null`?